ui/apps/platform/src/Containers/Policies/Wizard/Step1/MitreTechniqueSelect.tsx

```diff
@@ -24,52 +34,128 @@ function MitreTechniqueSelect({
+                        {selectedTechnique.id}
+                    </FlexItem>
+                </Flex>
+            );
+        }
+
+        return techniqueId;
+    };
+
+    const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+        <MenuToggle
+            ref={toggleRef}
+            onClick={onToggle}
+            isExpanded={isOpen}
+            aria-label={label}
+            className="pf-v5-u-w-100"
+        >
+            {getDisplayContent()}
+        </MenuToggle>
+    );
 
     return (
         <Select
             aria-label={label}
             className={className}
             isOpen={isOpen}
+            selected={techniqueId}
             onSelect={onSelect}
-            onToggle={(_event, val) => setIsOpen(val)}
-            placeholderText={label}
-            selections={techniqueId}
-            maxHeight="300px"
+            onOpenChange={(nextOpen: boolean) => setIsOpen(nextOpen)}
+            toggle={toggle}
+            shouldFocusToggleOnSelect
         >
-            {mitreTechniques.map(({ id, name }) => {
-                const optionClassName = id.includes('.')
-                    ? 'mitre-subtechnique-option'
-                    : 'mitre-technique-option';
-                const indexOfColonSpace = name.indexOf(': ');
-
-                // See MitreAttackVectorsFormSection.css for name and id style rules.
-                return (
-                    <SelectOption
-                        key={id}
-                        value={id}
-                        isDisabled={getIsDisabledOption(id)}
-                        className={optionClassName}
-                    >
-                        <Flex
-                            flexWrap={{ default: 'nowrap' }}
-                            justifyContent={{ default: 'justifyContentSpaceBetween' }}
-                        >
-                            <span className="name">
-                                {indexOfColonSpace === -1
-                                    ? name
-                                    : name.slice(indexOfColonSpace + 2)}
-                            </span>
-                            <span className="id">{id}</span>
-                        </Flex>
-                    </SelectOption>
-                );
-            })}
+            <SelectList style={{ maxHeight: '300px', overflowY: 'auto' }}>
+                {Object.entries(groupedTechniques).map(([baseId, techniques], index, array) => {
+                    // Sort techniques within group: base technique first, then sub-techniques
+                    const sortedTechniques = techniques.sort((a, b) => {
+                        if (a.id === baseId) {
+                            return -1;
+                        }
+                        if (b.id === baseId) {
+                            return 1;
+                        }
+                        return a.id.localeCompare(b.id);
+                    });
```

Comment:
Nit: Could the sorting be done at the same time as the grouping to simplify the logic in here?